### ISSUE TITLE
fix(filters): 2D/3D `exponential_filter` normalises by the corner wavenumber, leaving axis-Nyquist modes almost undamped

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#89]
Part of #80.

## Summary

`SpectralFilter2D.exponential_filter` and `SpectralFilter3D.exponential_filter` build `exp(−α (|k| / k_max)^p)` with `k_max = sqrt(K2).max()`, i.e. the **corner** of the wavenumber box. The axis Nyquist mode then sits at ratio `1/√2` (2D) or `1/√3` (3D) instead of 1, so with the defaults `α = 36, p = 16` it is damped by `exp(−36 · 2⁻⁸) ≈ 0.87` rather than `exp(−36) ≈ 2e-16`. On anisotropic grids it is worse: on 64×16 the short axis's Nyquist sits at ratio 0.24 and is damped by 0.99999999. The 1D filter is correct.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp
from spectraldiffx import FourierGrid2D, SpectralFilter2D

g = FourierGrid2D.from_N_L(Nx=64, Ny=64, Lx=2 * jnp.pi, Ly=2 * jnp.pi)
u = jnp.cos(32 * g.x)[None, :] * jnp.ones((64, 1))   # x-Nyquist checkerboard
out = SpectralFilter2D(grid=g).exponential_filter(u)
print(float(jnp.linalg.norm(out) / jnp.linalg.norm(u)))   # 0.8688  (expected ~1e-16)
```

Filter values at the axis Nyquist:

```
1D  N=64:        F(Nyq)            = 2.3e-16   (correct)
2D  64x64:       F(kx=Nyq, ky=0)   = 0.8688
2D  64x16:       F(ky=Nyq, kx=0)   = 0.99999999
3D  32x32x32:    F(axis Nyq)       = 0.9945
```

## Root cause

- `spectraldiffx/_src/fourier/filters.py:116-121` (2D): `k_mag = sqrt(K2); k_max = k_mag.max()`.
- `filters.py:158-163` (3D): same.

## Proposed fix

Use a tensor-product filter, consistent with the per-axis 2/3 mask the grid already uses:

```python
F = exp(-α (|kx|/kx_max)^p) * exp(-α (|ky|/ky_max)^p) [* exp(-α (|kz|/kz_max)^p)]
```

with `k*_max` the per-axis Nyquist. Alternative: keep the isotropic form but normalise by `min(kx_max, ky_max, ...)` and clip the ratio to 1. The tensor-product form is the standard choice (Hou & Li 2007 style filters are applied per axis) and is what the 2/3 rule assumes.

The same corner normalisation pattern appears in `SpectralFilter2D/3D.hyperviscosity` if it uses a normalised `K2`; check while there. The 7 copies of the filter formula across Fourier/Chebyshev/spherical are consolidated in #83; this issue only fixes the 2D/3D Fourier ones.

## Acceptance criteria

- [ ] Test: `F(axis Nyquist) == exp(−α)` to 1e-12 on 64×64, 64×16 and 32×32×32 grids.
- [ ] Test: a pure checkerboard along one axis is removed to < 1e-12 relative norm after one application.
- [ ] Test: the filter preserves the mean and satisfies `F(F(u)) == F²(u)` (idempotence of the multiplier).
- [ ] Docstring states the tensor-product form.
